File: domox-webapp/src/main/java/domox/webapp/Application.java
package domox.webapp;

import org.apache.causeway.core.config.presets.CausewayPresets;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Import;

@SpringBootApplication
@Import({AppManifest.class})
public class Application /*extends SpringBootServletInitializer*/ {

    /**
     * @implNote this is to support the <em>Spring Boot Maven Plugin</em>, which auto-detects an
     * entry point by searching for classes having a {@code main(...)}
     */
    public static void main(String[] args) {
        CausewayPresets.prototyping();
        SpringApplication.run(Application.class, args);
    }

}

File: module-ksimple/src/main/kotlin/org.apache.isis.domox/modules/ksimple/Application.kt
package org.apache.isis.domox.modules.ksimple

import org.springframework.beans.factory.annotation.Configurable
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Import

@Configurable
@SpringBootApplication
@Import()
open class Application {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(Application::class.java, *args)
        }
    }
}
